framework/test/gtest/ServiceListenerTest.cpp

```diff
@@ -0,0 +1,379 @@
+
+  //Check first service listener
+  ASSERT_TRUE(listener1.checkEvents(events));
+  //Check second service listener
+  ASSERT_TRUE(listener2.checkEvents(events));
+
+  context.RemoveServiceListener(&listener1,
+                                &TestServiceListener::serviceChanged);
+  context.RemoveServiceListener(&listener2,
+                                &TestServiceListener::serviceChanged);
+}
+
+TEST_F(ServiceListenerTest, frameSL05a)
+{
+  std::vector<ServiceEvent::Type> events;
+  events.push_back(ServiceEvent::SERVICE_REGISTERED);
+  events.push_back(ServiceEvent::SERVICE_UNREGISTERING);
+
+  auto bundle = InstallLib(framework.GetBundleContext(), "TestBundleA");
+
+  bool testStatus = runStartStopTest(
+    "FrameSL05a", 1, bundle, framework.GetBundleContext(), events);
+
+  ASSERT_TRUE(testStatus);
+}
+
+TEST_F(ServiceListenerTest, frameSL10a)
+{
+  std::vector<ServiceEvent::Type> events;
+  events.push_back(ServiceEvent::SERVICE_REGISTERED);
+  events.push_back(ServiceEvent::SERVICE_UNREGISTERING);
+
+  auto bundle = InstallLib(framework.GetBundleContext(), "TestBundleA2");
+
+  bool testStatus = runStartStopTest(
+    "FrameSL10a", 1, bundle, framework.GetBundleContext(), events);
+  ASSERT_TRUE(testStatus);
+}
+
+TEST_F(ServiceListenerTest, frameSL25a)
+{
+  auto context = framework.GetBundleContext();
+
+  TestServiceListener sListen(context, false);
+  EXPECT_NO_THROW(
+    context.AddServiceListener(&sListen, &TestServiceListener::serviceChanged))
+    << "service listener registration failed ";
+
+  auto libSL1 = InstallLib(context, "TestBundleSL1");
+  auto libSL3 = InstallLib(context, "TestBundleSL3");
+  auto libSL4 = InstallLib(context, "TestBundleSL4");
+
+  std::vector<ServiceEvent::Type> expectedServiceEventTypes;
+
+  // Startup
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_REGISTERED); // at start of libSL1
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_REGISTERED); // FooService at start of libSL4
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_REGISTERED); // at start of libSL3
+
+  // Stop libSL4
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_UNREGISTERING); // FooService at first stop of libSL4
+
+#ifdef US_BUILD_SHARED_LIBS
+  // Shutdown
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_UNREGISTERING); // at stop of libSL1
+  expectedServiceEventTypes.push_back(
+    ServiceEvent::SERVICE_UNREGISTERING); // at stop of libSL3
+#endif
+
+  // Start libBundleTestSL1 to ensure that the Service interface is available.
+  EXPECT_NO_THROW(libSL1.Start()) << "Failed to load bundle";
+
+  // Start libBundleTestSL4 that will require the serivce interface and publish
+  // cppmicroservices::FooService
+  EXPECT_NO_THROW(libSL4.Start()) << "Failed to load bundle";
+
+  // Start libBundleTestSL3 that will require the serivce interface and get the service
+  EXPECT_NO_THROW(libSL3.Start()) << "Failed to load bundle";
```

Comment:
consistency; `EXPECT_NO_THROW` is used but not consistently in this file or in `usResourceCompilerTest.cpp` (where the original test points checked for exceptions being thrown). 

I can understand not using the `*_NO_THROW` macro in all of those instances since an exception would fail the test and that is the desired outcome. I'd suggest being consistent in the use of this type of test macro. It seems like the use of the macro is unnecessary so my suggestion would be to remove the use of it where it makes sense.